src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopExe.cs

```diff
@@ -830,6 +830,50 @@ public void It_places_package_satellites_correctly(bool crossTarget)
             outputDirectory.Should().HaveFile(@"fr\FluentValidation.resources.dll");
         }
 
+        [WindowsOnlyFact]
+        public void It_places_package_pdb_file_correctly()
+        {
+            var testAsset = _testAssetsManager
```

Comment:
Rather than checking in test projects into the `Assets` folder, if they are no super complicated I prefer creating them with the `TestProject` APIs.  See for example https://github.com/dotnet/sdk/blob/868f9e2b7494f1e1e44cd15da3d5741e1e27e4a4/src/Tests/Microsoft.NET.Build.Tests/AppHostTests.cs#L307-L318

The test projects should probably target `ToolsetInfo.CurrentTargetFramework`.

We should also be testing behavior for both build and publish.  I believe that the XML files should always be copied to the output directory for build, but only be copied to the publish directory if `PublishReferencesDocumentationFiles` is true.  So we should verify whether that's true (by checking the project to project reference behavior) and test the different cases.

Reply:
The last commit is using `TestProject ` apis in unit-tests